better-words/message_parser.py

```diff
@@ -0,0 +1,61 @@
+import yaml
+import re
+from os import listdir
+from os.path import splitext
+
+class MessageParser(object):
+  def __init__(self):
+    self.words = {}
+
+    # iterate over all the word categories and load up each yml file
+    for file in listdir('./better-words/words'):
+      category, extension = splitext(file)
+      if extension == '.yml':
+        with open('./better-words/words/' + file, 'r') as stream:
+          self.words[category] = yaml.load(stream)
+
+  # check if the given message contains a word or phrase match
+  # TODO: handle messages that contain multiple matches
+  # TODO: check if the user has opted out
+  def is_match(self, message):
+    for category in self.words:
+      for entry in self.words[category]['entries']:
+        for word in entry['words']:
+          regex = r'\b' + word + r'\b'
+          match = re.search(regex, message.get('text'))
+          if match:
+            return {
+              'message': message,
+              'category': self.words[category],
+              'entry': entry,
+              'word': match.group(0)
+            }
+    return None
+
+  # format a message to be sent to the user
+  def format(self, match):
+    message = match['message']
```

Comment:
afaik, this doesn't work for dictionary keys only objects. maybe am doing something wrong. spent some time looking through this SO post yesterday trying to figure out how to make it work https://stackoverflow.com/questions/2352181/how-to-use-a-dot-to-access-members-of-dictionary